doc/HowToUseMeson.md

```diff
@@ -0,0 +1,234 @@
+There should be declared:
+1. Project name,
+2. Programming languages(meson trying to find a suitable compiler for named languages ),
+3. Suitable meson version,
+4. App version,
+5. Licenses,
+6. Etc.
+
+Cython files compiles in order: `.pyx` &rarr; `.c` or `.cpp` &rarr; `.so`.
+So there must be declared at least 2 languages: `cpp` or `c` and `cython`.
+
+
+### <a name="subparagraph22"></a> Useful system checks 
+For apps which are using `cython` can be added manual check
+using `find_program` function:
+```meson
+cython = find_program('cython', required : true)
+if not cython.found()
+  error('MESON_BUILD_FAILED: Cython3 not found.')
+endif
+```
+
+Also, the `python` installation must be checked too. 
+Default `pyhton` dependencies saved into variable
+to insert them into every compile target
+```meson
+py_mod = import('python')
+py3 = py_mod.find_installation()
+py3_dep = py3.dependency()
+```
+As for `MC_lib`, the most part of it uses `numpy` API,
+so numpy declared as dependency, by its location:
+```meson
+incdir_numpy = run_command(py3,
+  ['-c', 'import os; os.chdir(".."); import numpy; print(numpy.get_include())'],
+  check : true
+).stdout().strip()
+
+inc_np = include_directories(incdir_numpy)
+```
+
+
+## <a name="paragraph3"></a> Meson + Python 
+There is no need to compile pure `python` files,
+so all of them can be installed as sources:
+```meson
+python_sources = [
+    '__init__.py',
+    'Pytester.py',
+] # declare list of .py files
+
+py3.install_sources(
+    python_sources,
+    pure: false,
+    subdir: install_dir
+)
+```
+
+
+## <a name="paragraph4"></a> Meson + Cython 
+To make compiled `cython` file accessible
+to its `.pxd` (cython API) or `.py` (python API), firstly
+it must compiled into `.c` or `.cpp` file then 
+it must be converted into `shared_library` meson class object.
+
+Compile cpp file
+Meson version < 0.59:
+```meson
+_observable_cpp = custom_target('observable_cpp',
+  output : 'observable.cpp',
+  input : 'observable.pyx',
+  command : [cython, '--cplus', '-3', '--fast-fail', '@INPUT@', '-o', '@OUTPUT@']
+)
+
+py3.extension_module(
+  'observable', _observable_cpp,
+  include_directories: inc_np,
+  dependencies : [py3_dep],
+  install: true,
+  subdir: install_dir
+)
+```
+---------
+
+###Maybe will work later
+
+Meson version > 0.59:
+
+```meson
+py3.extension_module(
+  'observable', _observable_cpp,
+  include_directories: inc_np,
+  dependencies : py3_dep,
+  install: true,
+  subdir: install_dir,
+  cython_args: '--cplus'
+)
+```
+
+As for my work, compiler argument  `cython_args: '--cplus'`
+don't work, meson do not apply any `cython` args, which I added
+by this command. Also using `custom target` give developer a possibility
+to control `.so` files compilation. For example, to build 
+MC_lib on OSX it necessary to use `C++11`, and using `custom target`
+for `.pyx` &rarr; `.cpp` take positional args for `cython` compiler and
+'extension_module' take positional args for `C++` compiler
```

Comment:
`extension_module` you can give `c_args` and they will be picked up if the input is `.pyx` and the target language C